File: python/a05.py
# A05 Three Cards
# submission: https://atcoder.jp/contests/tessoku-book/submissions/38326350

n, k = map(int, input().split())
ans = 0

for r in range(1, n + 1):
    for b in range(1, n + 1):
        w = k - r - b
        if 1 <= w <= n:
            ans += 1

print(ans)

File: rust/tessoku-book/src/bin/a05.rs
// A05 Three Cards
// submission: https://atcoder.jp/contests/tessoku-book/submissions/38365578

use proconio::input;

fn main() {
    input!{
        n: i32,
        k: i32,
    }
    let mut ans: u32 = 0;
    for r in 1..=n  {
        for b in 1..=n {
            let w = k - r - b;
            if 1 <= w && w <= n {
                ans += 1;
            }
        }
    }
    println!("{}", ans);
}
